Packs/MS-ISAC/Integrations/MSISAC/MSISAC.py

```diff
@@ -201,39 +240,103 @@ def retrieve_events_command(client: Client, args: Dict[str, Any]):
     )
 
 
+@logger
+def fetch_incidents(client: Client, first_fetch: datetime, last_run: Dict) -> tuple[List[dict[str, Any]], Dict]:
+    """Uses to fetch events into XSIAM
+    Args:
+        client: Client object with request
+        first_fetch: String from when to fetch if first time. "%Y-%m-%dT%H:%M:%SZ"
+        last_run: Last fetch object occurs.
+    Returns:
+        incidents, new last_run
+    """
+
+    fetch_time: datetime
+
+    if not last_run.get("lastRun"):
+        fetch_time = first_fetch
+    else:
+        fetch_time = datetime.strptime(last_run.get("lastRun", ""), XSOAR_INCIDENT_DATE_FORMAT)
+
+    fetch_time_lookback_days: int = calculate_lookback_days(fetch_time, datetime.now())
+
+    retrieve_events_data: dict = client.retrieve_events(days=fetch_time_lookback_days).get("data", [])
+
+    events_to_fetch: list[dict] = []
+    latest_event_s_time: datetime = fetch_time
+
+    # API returns a list if there is albert event data. data key is a string if there is no data.
+    if isinstance(retrieve_events_data, list):
+        for event in retrieve_events_data:
+            event_s_time = datetime.strptime(event.get("stime"), MSISAC_S_TIME_FORMAT)
+            event_id = event.get("event_id", "")
+            event_description = event.get("description", "")
+            if event_s_time > fetch_time:  # Make sure event happened after last fetch
+                events_to_fetch.append(
+                    {
+                        "name": f"{event_id} - {event_description}",
+                        "occurred": event_s_time.strftime(XSOAR_INCIDENT_DATE_FORMAT),
+                        "rawJSON": json.dumps(event),
+                        # We are not using mirroring.
+                        # This will show ingested event numbers in fetch history modal.
+                        "dbotMirrorId": f"{event_id}",
+                    }
+                )
+                if event_s_time > latest_event_s_time:
+                    latest_event_s_time = event_s_time
+
+    else:
+        demisto.debug(f"Here is the event data that was returned: {retrieve_events_data}")
+
+    next_run_dict = {"lastRun": latest_event_s_time.strftime(XSOAR_INCIDENT_DATE_FORMAT)}
+
+    return events_to_fetch, next_run_dict
+
+
 """ MAIN FUNCTION """
 
 
 def main():
-    api_key = demisto.params().get("apikey")
+    params = demisto.params()
+    args = demisto.args()
+    command = demisto.command()
 
-    base_url = urljoin(demisto.params()["url"], "/api/v1")
+    api_key = params.get("apikey", {}).get("credentials", {}).get("sshkey", "")
```

Comment:
```suggestion
    api_key = params.get("apikey", {}).get("credentials", {}).get("sshkey", "") or params.get("apikey", {}).get("password", "")
```
To avoid breaking backwards compatibility.

Reply:
I was curious if that was going to break something, since I changed around the parameter. The old code goes directly to the "apikey" param. I imagine it would be best to add the or clause, and drop off the .get("password") portion.
<img width="381" alt="image" src="https://github.com/user-attachments/assets/6fb3665e-ac47-4f2d-bc4c-d2d3823f64af" />